src/cockpit/channels/dbus.py

```diff
@@ -172,11 +171,54 @@ class DBusChannel(Channel):
     matches = None
     name = None
     bus = None
+    owner = None
 
     # This needs to be a fair mutex so that outgoing messages don't
     # get re-ordered.  asyncio.Lock is fair.
     watch_processing_lock = asyncio.Lock()
 
+    def add_match(self, rule, handler):
+        def sync_handler(message):
```

Comment:
I don't like this at all.  Handlers should be simple callbacks.

Reply:
This is old code that just got moved around. I'll try to improve this in another PR.